### DISCUSSION TITLE
Evaluation: memory issues

### DISCUSSION THREAD

**Discussion by hp52** [explosion/spaCy#7874] [Help: Coding & Implementations] [answered]
I'm trying to evaluate a trained multilabel textcat model:

`python -m spacy evaluate model-best test.spacy -g 0`

It crashes with a memory problem:

`RuntimeError: CUDA out of memory. Tried to allocate 154.00 MiB (GPU 0; 7.93 GiB total capacity; 5.92 GiB already allocated; 146.06 MiB free; 7.01 GiB reserved in total by PyTorch)`

What can I do in this case to fix the memory issue? Can I specify a smaller batch size anywhere?

Thanks for your help.

**Comment by adrianeboyd** [comment, marked as answer]
There's a `batch_size` parameter as `[nlp.batch_size]` in the model config that you can modify. You can edit `model-best/config.cfg` by hand to modify for use with `spacy evaluate`. In a script, you can provide a `batch_size` kwarg with `.pipe` or `.evaluate` if you want to override the value from the config.

**Reply by hp52** [reply]
Thanks for your reply. In my config file, the batch_size is in this section:
```
[nlp]
lang = "de"
pipeline = ["transformer","textcat_multilabel"]
batch_size = 64
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}
```
Is that the correct batch_size? I tried to modify it in the command
`python -m spacy evaluate model-best test.spacy -g 0 --nlp.batch_size 32`
Unfortunately, it didn't work, it tells me there's no such option. Also, I didn't understand the .pipe/.evaluate thing.
What am I doing wrong?

**Reply by adrianeboyd** [reply]
Only the commands that take `config.cfg` as an argument support the overrides like this. You will have to edit the config by hand for `spacy evaluate`.

In a script:

```python
nlp = spacy.load("model")
scores = nlp.evaluate(examples, batch_size=16)
```

You can also just copy the `spacy evaluate` script from `spacy/cli/evaluate.py`, adjust the relative imports to absolute `spacy.` imports, and run it as a standalone script with your own edits/args.
